### PR TITLE
Replaced some deprecated methods with suggested alternatives

### PR DESCRIPTION
**Pull request by StefanSpieker**
Replaced some deprecated methods with suggested alternatives:
* `getTemporaryFile()` with `getTemporaryPath().toFile()`
* `Util.copyFile()` with `Files.copy()` with the `REPLACE_EXISTING` flag, since it was also overwriting before

### Proposed changelog entries

* N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content-Security-Policy directives (see [documentation on jenkins.io](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6980"></a>

### CODE REVIEW

File: core/src/main/java/hudson/model/AbstractItem.java
Comment (basil) [inline core/src/main/java/hudson/model/AbstractItem.java:897]: Is this really buying us anything other than moving the `#toFile` invocation from `AtomicFileWriter` to `AbstractItem`? Either way it's the same logic as before. I don't feel strongly about keeping this change, but I also don't feel strongly about undoing it either. Up to you.

File: core/src/main/java/hudson/triggers/SCMTrigger.java
Comment (basil) [inline core/src/main/java/hudson/triggers/SCMTrigger.java:539]: Consistent with `new AnnotatedLargeText<>(getLogFile(), Charset.defaultCharset()` a few lines below (and `Files.readString(Util.fileToPath(logFile), Charset.defaultCharset())` some number of lines after that), and definitely a step in the right direction, but not complete. For completeness the other calls to `new StreamTaskListener(getLogFile())` in this class should also be updated to use the two-argument version that takes a `Charset` as the second argument. I wouldn't block this PR over it, but I would definitely prefer the more complete version if you have the time.

File: test/src/test/java/hudson/model/ProjectTest.java
Comment (basil) [inline test/src/test/java/hudson/model/ProjectTest.java:526]: In situations like this I tend to prefer the more specific `UncheckedIOException`. In general this is considered good coding style; see e.g. [2. Prefer Specific Exceptions](https://stackify.com/best-practices-exceptions-java/).

**Review by basil — COMMENTED** [review]
Looks good overall; I left some optional feedback for your consideration. If you're interested in taking this a little further, great! If not, I can approve this once the tests are passing.

File: core/src/main/java/hudson/triggers/SCMTrigger.java
Comment (StefanSpieker) [reply core/src/main/java/hudson/triggers/SCMTrigger.java:539]: Yes you are right, added the `Charset.defaultCharset()` also to these two missing calls.

File: test/src/test/java/hudson/model/ProjectTest.java
Comment (StefanSpieker) [reply test/src/test/java/hudson/model/ProjectTest.java]: Thanks for the suggestion and also the link for clarification.

File: core/src/main/java/hudson/model/AbstractItem.java
Comment (StefanSpieker) [reply core/src/main/java/hudson/model/AbstractItem.java:897]: The `getTemporaryFile()` is deprecated so I thought we can replace it with the suggested one.

**Review by basil — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
